t/cobrand/aberdeenshire.t

```diff
@@ -176,6 +176,437 @@ FixMyStreet::override_config {
 
             $mech->clear_emails_ok;
         };
+
+        subtest 'open311_get_update_munging handles defect superseding' => sub {
+            my $cobrand = FixMyStreet::Cobrand::Aberdeenshire->new;
+
+            # Create a superseded report with external_id starting with DEFECT_
+            my ($superseded_report) = $mech->create_problems_for_body(1, $aberdeenshire->id, 'Superseded Report', {
+                category => 'Pothole',
+                cobrand => 'aberdeenshire',
+                latitude => 57.28126,
+                longitude => -2.44012,
+                areas => '2648',
+                external_id => 'DEFECT_12345',
+                state => 'confirmed'
+            });
+
+            # Create a new superseding report
+            my ($superseding_report) = $mech->create_problems_for_body(1, $aberdeenshire->id, 'Superseding Report', {
+                category => 'Pothole',
+                cobrand => 'aberdeenshire',
+                latitude => 57.28126,
+                longitude => -2.44012,
+                areas => '2648',
+                external_id => 'DEFECT_67890',
+                state => 'confirmed'
+            });
+
+            # Create some users who have alerts on the superseded report
+            my $alert_user1 = $mech->create_user_ok('alertuser1@example.com', name => 'Alert User 1');
+            my $alert_user2 = $mech->create_user_ok('alertuser2@example.com', name => 'Alert User 2');
+
+            my $alert1 = FixMyStreet::DB->resultset('Alert')->create({
+                user => $alert_user1,
+                alert_type => 'new_updates',
+                parameter => $superseded_report->id,
+                confirmed => 1,
+                cobrand => 'aberdeenshire',
+                lang => 'en-gb',
+                cobrand_data => '',
+                whensubscribed => DateTime->now->subtract(days => 1),
+            });
+
+            my $alert2 = FixMyStreet::DB->resultset('Alert')->create({
+                user => $alert_user2,
+                alert_type => 'new_updates',
+                parameter => $superseded_report->id,
+                confirmed => 1,
+                cobrand => 'aberdeenshire',
+                lang => 'en-gb',
+                cobrand_data => '',
+                whensubscribed => DateTime->now->subtract(days => 1),
+            });
+
+            # Create a comment/update for the superseding report that indicates superseding
+            my $superseding_comment = $mech->create_comment_for_problem(
```

Comment:
I guess to be a true copy of what is happening, the comment at this point in the flow isn't in the database yet (not that I think it matters, but I guess in theory something could try and look at an ID or something?).